pkg/flow/tracer_map.go

```diff
@@ -115,5 +116,6 @@ func (m *MapTracer) evictFlows(ctx context.Context, forwardFlows chan<- []*Recor
 	default:
 		forwardFlows <- forwardingFlows
 	}
+	runtime.GC() // Triggers a manual GC
```

Comment:
I'm not sure about this manual call. For me the `GOMEMLIMIT` is supposed to be good enough if properly set.

Would it be interesting to make this configurable ? 
We can expose it in [debug](https://github.com/netobserv/network-observability-operator/blob/main/api/v1beta1/flowcollector_types.go#L222) section of the CRD.

Reply:
manual call because under normal run not close to resource limit GOMEMLIMIT won't trigger GC so we will keep leaking flows map memory causing the memory to keep building up this manual call to prevent this memory buildup IMO

Reply:
I agree with Julien, we know it has a CPU cost, having memory piling up is not bad per se as long as it's in the limit bounds configured.
GOMEMLIMIT should be sufficient to avoid OOM due to lack of gc calls, per my understanding.
But +1 to keep this option configurable so that we can still have a more aggressive GC strategy that we can turn on and off any time.

Reply:
added envvar to control it and default it to true because we saw immediate memory gain with it at the cost of cpu pump I will explore options to better trim down those resources in the following PRs